src/Parser/VengeanceDemonHunter/CombatLogParser.js

```diff
@@ -89,23 +91,28 @@ class CombatLogParser extends MainCombatLogParser {
   }
 
   generateResults() {
+      
     const results = super.generateResults();
 
     const fightDuration = this.fightDuration;
 
-    const nonDpsTimePercentage = this.modules.alwaysBeCasting.totalDamagingTimeWasted / fightDuration;
     const deadTimePercentage = this.modules.alwaysBeCasting.totalTimeWasted / fightDuration;
 
-    console.log(this.modules.alwaysBeCasting.totalTimeWasted+'---'+fightDuration);
+    // As soon as the information is ready to be analysed, gets it and put it in variables
+    // This is done to not get the 'cannot read property of undefined' error
+    if(this.modules.abilityTracker.abilities[SPELLS.SOUL_FRAGMENT.id] !== undefined)  {
```

Comment:
The 'undefined' value has the same "meaning" as 'false' in if's like that one? Didn't knew that for sure XD